src/api/course/index.ts

```diff
@@ -84,23 +94,137 @@ export class CourseApi {
       if (storeData) {
         this.emitCourseInfo(true)
       }
+      // register delta handler
+      ;(this.app as any).registerDeltaInputHandler(this.deltaHandler)
       resolve()
     })
   }
 
+  /** handle and process incoming course stream data */
+  private deltaHandler = (
+    delta: ValuesDelta,
+    next: (delta: ValuesDelta) => void
+  ) => {
+    this.processStreamSources(delta)
+    next(delta)
+  }
+
+  /** Process stream <destination>.nextPoint data
+   * Note: Stream source cannot override destination set by API!
+   * Destination is set when:
+   * 1. There is no current destination
+   * 2. msg source matches current Destination source
+   * 3. Destination Position is changed.
+   */
+  private async processStreamSources(delta: ValuesDelta) {
+    if (!this.cmdSource || (this.cmdSource && this.cmdSource.type !== 'API')) {
+      delta.updates.forEach((update: Update) => {
+        if (
+          !update.values ||
+          (update.values && !Array.isArray(update.values))
```

Comment:
I think `!Array.isArray(update.values)` covers all the cases and conveys the intent.